src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -2544,7 +2545,7 @@ private void LexConflictMarkerHeader(ref SyntaxListBuilder triviaList)
             this.AddTrivia(SyntaxFactory.ConflictMarker(TextWindow.GetText(false)), ref triviaList);
         }
 
-        private void AddTrivia(CSharpSyntaxNode trivia, ref SyntaxListBuilder list)
+        private void AddTrivia(CSharpSyntaxNode trivia, [NotNull] ref SyntaxListBuilder? list)
```

Comment:
i'm a bit confused by this annotation.

Reply:
Check out the body of the method. It accepts null, but if it's null, it will make it not null. `[NotNull]` means the output is never null.

Reply:
Ah, gotcha :)